### PR TITLE
Various Fixes

### PR DESCRIPTION
**Pull request by Shadowghost**
* add locking when processing user data
* fixes `ids` spelling
* fixes wrongly configured logger in the sync task
* remove trakt API cache walkaround ([issue has been resolved](https://github.com/trakt/api-help/discussions/341))
* add handling of additional status codes to `RetryHttpRequest` based on [trak.tv API docs](https://trakt.docs.apiary.io/#introduction/status-codes)

### CODE REVIEW

File: Trakt/Configuration/PluginConfiguration.cs [hunk 1]
```diff
@@ -46,8 +45,21 @@ public void AddUser(string userId)
...
+
+    /// <summary>
+    /// Gets a list of all trakt users.
+    /// </summary>
+    /// <returns>IReadonlyList{TraktUser} with all trakt users.</returns>
+    public IReadOnlyList<TraktUser> GetAllTraktUsers()
+    {
+        lock (_traktUsers)
+        {
+            return _traktUsers.AsReadOnly();
```
Comment (cvium) [inline Trakt/Configuration/PluginConfiguration.cs:62]: Nothing really stops another thread from modifying the list after you return here.

Comment (Shadowghost) [reply Trakt/Configuration/PluginConfiguration.cs:62]: That's correct but I'm not sure how this influences the current code?

Comment (cvium) [reply Trakt/Configuration/PluginConfiguration.cs:62]: Why have the lock at all?

File: Trakt/Configuration/PluginConfiguration.cs [hunk 2]
```diff
@@ -26,18 +29,20 @@ public PluginConfiguration()
     public TraktUser[] TraktUsers { get; set; }
```
Comment (Shadowghost) [inline Trakt/Configuration/PluginConfiguration.cs:29]: I can't remove the public array because otherwise the frontend has no access to this...

File: Trakt/Helpers/UserDataManagerEventsHelper.cs [hunk 3]
```diff
@@ -159,12 +159,28 @@ public void ProcessUserDataSaveEventArgs(UserDataSaveEventArgs userDataSaveEvent
...
+        lock (_userDataPackages)
+        {
+            if (!_userDataPackages.Any())
+            {
+                _logger.LogInformation("No events... stopping queue timer");
+                return;
+            }
+
+            userDataQueue = _userDataPackages;
+            _userDataPackages.Clear();
```
Comment (cvium) [inline Trakt/Helpers/UserDataManagerEventsHelper.cs:180]: Does this work? You're just saving a reference to a local variable and then clearing the dictionary. `userDataQueue` will be empty then?

File: Trakt/Configuration/PluginConfiguration.cs (hunk 2 above)
Comment (cvium) [reply Trakt/Configuration/PluginConfiguration.cs:29]: But you could just use `_traktUsers` as the backing field and remove the setter?

Comment (Shadowghost) [reply Trakt/Configuration/PluginConfiguration.cs:29]: I can't because otherwise the plugin settings are not persisted on disk. Only public fields are saved.

Comment (Shadowghost) [reply Trakt/Configuration/PluginConfiguration.cs:27]: I removed the locking because saving the plugin settings via the plugin config server endpoint makes the locking useless.

File: Trakt/Helpers/UserDataManagerEventsHelper.cs (hunk 3 above)
Comment (Shadowghost) [reply Trakt/Helpers/UserDataManagerEventsHelper.cs]: You're correct, I created a new dict from the current one that seems to work.

File: Trakt/Configuration/PluginConfiguration.cs [hunk 4]
```diff
@@ -50,4 +51,13 @@ public void RemoveUser(string userId)
...
         TraktUsers = traktUsers.ToArray();
     }
+
+    /// <summary>
+    /// Gets a list of all trakt.tv users.
+    /// </summary>
+    /// <returns>IReadonlyList{TraktUser} with all trakt users.</returns>
+    public IReadOnlyList<TraktUser> GetAllTraktUsers()
+    {
+        return TraktUsers.ToList().AsReadOnly();
```
Comment (crobibero) [inline Trakt/Configuration/PluginConfiguration.cs:61]:
```suggestion
        return TraktUsers.ToList();
```
not needed

**Comment by crobibero** [comment]
Now that you're working on thread safety I don't think `ProcessUserDataSaveEventArgs` is safe like you want it to be- if the same user changes data in quick succession there's a chance that only the second will be written.

tldr I think the entire function needs a lock?

**Comment by Shadowghost** [comment]
Not sure how you'd do that since it's just a callback for the event listener on user data changes.

**Comment by crobibero** [comment]
Yes, that should do the trick
